src/DeleteCapablePdoTrait.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Expression\LogicalExpressionInterface;
+use Dhii\Expression\TermInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use PDOStatement;
+
+/**
+ * Common functionality for objects that can delete records in a database via PDO.
+ *
+ * @since [*next-version*]
+ */
+trait DeleteCapablePdoTrait
+{
+    /**
+     * Executes a DELETE SQL query, deleting records in the database that satisfy the given condition.
+     *
+     * @since [*next-version*]
+     *
+     * @param LogicalExpressionInterface|null $condition Optional condition that records must satisfy to be deleted.
+     *
+     * @return PDOStatement The executed PDO statement.
+     */
+    protected function _delete(LogicalExpressionInterface $condition = null)
+    {
+        $fieldNames   = $this->_getSqlDeleteFieldNames();
```

Comment:
Also, why are they ignored when building the hash map?

Reply:
The return value is documented in the doc for `_getSqlDeleteFieldNames()` as `string[]|Stringable[] A list of field names`. Is this unclear?

They are ignored because the value-hash map is for **values**. Field names are not values. Consider an expression that would be rendered into `WHERE name = "duck"`. To invoke this query with PDO, only `"duck"` needs to be extracted and replaced with a hash.

Reply:
I understand what the _type_ is. But I don't understand *what* it is, why it is there, what purpose it serves, and how "field names" can be matched with the idea of "deleting" something.

Reply:
A query like this:

```
DELETE FROM `t` WHERE `name` = "duck" OR `name` = "duckling";
```

Needs to be transformed into this:

```
DELETE FROM `t` WHERE `name` = :12345 OR `name` = :67890;
```

Since the WHERE clause is expressed as a logical expression, the expression must be traversed, and each value encountered must have a hash generated for it. This is what the value-hash map is. Since expressions are immutable, the expression itself is not changed - but the value-hashmap is later given to the expression renderer.

Generating the value-hashmap is not so straight-forward, because expressions can be literally anything. Therefore, the algorithm accepts a list of strings to ignore. In this case, things like "name" are ignored. Because otherwise, this would be the resulting query, which is incorrect:

```
DELETE FROM `t` WHERE :123abc = :345def OR :567abc = :789def;
```